File: Langauge/Java/Reactive/02_Reactor/webfluxTuto/src/main/java/com/example/webfluxtuto/controller/MemberController.java
package com.example.webfluxtuto.controller;


import com.example.webfluxtuto.dto.MemberRequest;
import com.example.webfluxtuto.dto.MemberResponse;
import com.example.webfluxtuto.model.Member;
import com.example.webfluxtuto.service.MemberService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/members")
public class MemberController {
    private final MemberService memberService;

    @GetMapping
    public Mono<List<MemberResponse.Member>> findAll(@Valid Mono<MemberRequest.Member> member) {
        return memberService.findAll(member);
    }

    @PostMapping
    public Mono<MemberResponse.Member> insert(@RequestBody Mono<MemberRequest.Member> member) {
        return memberService.insert(member);
    }

    @DeleteMapping
    public Mono<MemberRequest.MemberIdList> delete(Mono<MemberRequest.MemberIdList> members) {
        return memberService.delete(members);
    }


}

File: Langauge/Java/Framework/mybatis/basicRetry/demo/src/main/kotlin/com/example/demo/controller/MemberController.kt
package com.example.demo.controller

import com.example.demo.entity.Member
import com.example.demo.service.MemberService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.Mapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/members")
class MemberController(
    private val memberService: MemberService
) {

    @GetMapping("/{id}")
    fun selectById(@PathVariable id: Long): ResponseEntity<Member> {
        return ResponseEntity.ok()
            .body(memberService.selectById(id))
    }
    @GetMapping
    fun selectAll(): ResponseEntity<List<Member>> {
        return ResponseEntity.ok()
            .body(memberService.selectAll())
    }
}